Feature/artciles: add softbreak plugin

maerzhase opened the pull request:
-  fix #304

ciphrd committed:
Merge pull request #289 from fxhash/feature/articles-tags

feat(article): add tags to article; fix(article): fix display image in preview display correctly

GemN committed:
refactor(gtk): refacto sort and filters

GemN committed:
fix(gtk): add back scroll top on search

ciphrd committed:
🐛 don't spread lists on umtiline when instanciating

ciphrd committed:
Merge pull request #293 from fxhash/feature/articles-list-flatten

feat(article): unwrap listItem for slate

ciphrd committed:
Merge pull request #295 from fxhash/feature/articles-search

feat(articles): add sort and search to article

ciphrd committed:
✨ added simple code editor & prism for syntax highlighting

ciphrd committed:
💄 article code block viewer (with prism)

ciphrd committed:
✨ added constraints to the code node

ciphrd committed:
💄 style for inline code blocks

ciphrd committed:
Merge pull request #299 from fxhash/feature/article-code-editor

Articles, code block (edit & view)

ciphrd committed:
🩹 code indent normalized

ciphrd committed:
Merge pull request #260 from fxhash/fix/#231-normalize-figure-alt-text

Fix/#231 normalize figure alt text

ciphrd committed:
✨ draft articles can be removed

ciphrd committed:
Merge pull request #305 from fxhash/feature/articles-delete-draft

Draft articles can be removed

ciphrd committed:
📝 added documentation about articles

maerzhase committed:
add softbreak plugin; fix #304

GemN commented:
I think it's too permissive at the moment, you're not checking the node type, it could add undesired side effects 
```js
  editor.insertSoftBreak = () => {
    // check the node type is p / ul / ol etc
    // insert \n
    insertSoftBreak() // to allow other plugins to work if they manipulate soft break
  }
```

maerzhase commented:
sure we can add that, i am not sure if its necessary though. softlinebreak is only triggered on text nodes. that should already exclude all custom blocks. 

but we can also retrieve the node by cursor position and do a check for the type.

maerzhase committed:
add util to retrieve block elements

maerzhase committed:
make softLineBreak plugin extensible

maerzhase commented:
I included the changes proposed by @GemN 👍

ciphrd committed:
📝 added links to article specifications

ciphrd committed:
Merge pull request #309 from fxhash/features/articles-doc

Documentation for articles

ciphrd reviewed (changes requested):
I believe it's good to check for the block type, as we want to avoid any weird side effect.

There is an issue in the markdown transformation, this is the string which gets computed:

```md
This is a test&#x20;
We can have multiline now, great

Another test
```

(see here https://gateway.fxhash-dev2.xyz/ipfs/Qmcfk4M1jwkFiN3FhEw1m1PoZX5nqjSPKqGHxuFKAM7qgV)

Resulting in the viewer not displaying the multiline properly:

![image](https://user-images.githubusercontent.com/18054507/182613818-d2d7c0a1-c84c-4f02-a1a3-604f44a77148.png)

I believe the resulting markdown should be:

```md
This is a test
We can have multiline now, great

Another test
```

This is by default rendered like that by markdown viewers:
> This is a test We can have multiline now, great

As there is no support for backline in paragraphs. It needs to be added.

ciphrd committed:
🐛 wrong url pushed to router when previewing fresh article

ciphrd committed:
Merge pull request #310 from fxhash/feature/article-first-preview-404

Wrong url pushed to router when previewing fresh article

ciphrd committed:
🐛 navigation on click not working on article editor

ciphrd committed:
🐛 article preview, reducing splits keeps user not found with the API

ciphrd committed:
Merge pull request #311 from fxhash/bug/cant-preview-mint

Article preview, reducing splits keeps user not found with the API

ciphrd committed:
🚨 fix compiler warnings

maerzhase committed:
use bulletOther option to avoid comments between lists

- fix #296

GemN committed:
feat(article): insert paragraph after break for heading

GemN committed:
refactor(article): breaks are defined as a plugin and property in block definition, ListItem break transforms as a new paragraph when empty

GemN committed:
fix: typing

maerzhase committed:
Merge branch 'feature/articles' into fix/#232-add-shortcut-for-creating-links

GemN committed:
Merge branch 'feature/articles' of github.com:fxhash/fxhash-website into feature/articles-table-normalize

maerzhase committed:
remove HotKeyPlugin

maerzhase committed:
allow inlineMenu configuration in blockDefinitions

GemN committed:
Merge branch 'fix/#232-add-shortcut-for-creating-links' of github.com:fxhash/fxhash-website into feature/articles-table-normalize

GemN committed:
fix: only allow text in table

GemN committed:
feat(article): add hr, fix(article): fix autoformat inline select position

GemN committed:
fix(article): block menu hide correctly on firefox, block menu doesnt flash position on init

GemN committed:
fix: file bleeding

GemN committed:
fix: file bleeding

ciphrd committed:
Merge pull request #312 from fxhash/fix/#296-fix-html-comment-between-lists

use bulletOther option to avoid comments between lists

maerzhase committed:
remove encoded whitespaces from markdown

ciphrd committed:
Merge pull request #313 from fxhash/feature/articles-enter

feat(article): insert paragraph after break for heading

ciphrd committed:
Merge pull request #319 from fxhash/feature/articles-firefox

fix(article): block menu hide correctly on firefox, block menu doesnt flash position on init

ciphrd committed:
💄 article hr style

maerzhase committed:
Revert "remove encoded whitespaces from markdown"

This reverts commit 397f27ae3d27f1535a00733574fb4ce58f4deb88.

ciphrd committed:
♻️ use generic thematic break component

ciphrd committed:
Merge branch 'feature/articles' into feature/articles-hr

maerzhase commented:
the encoding comes from the mdast-util-to-markdown: https://github.com/syntax-tree/mdast-util-to-markdown/commit/108450c5e3dc013f8e9f7119b63ed8feddff7a7b

when i started to replace the encoded whitespaces it also breaks the newlines. @ciphrd i am not sure how to resolve this issue

maerzhase committed:
Merge branch 'feature/articles' into fix/#304-softbreak-plugin

maerzhase committed:
fix merge conflict

ciphrd committed:
Merge pull request #318 from fxhash/feature/articles-hr

feat(article): add hr, fix(article): fix autoformat inline select position

ciphrd committed:
🚸 improved autosave tooltip ui/ux

ciphrd committed:
Merge pull request #320 from fxhash/feature/articles-autosave-ui

Improved autosave tooltip ui/ux

ciphrd committed:
🚸 lower part of buttons can be clicked

ciphrd committed:
🚸 showutility buttons when element is focused

ciphrd committed:
Merge pull request #321 from fxhash/feature/articles-always-buttons

Show buttons when block is focused

ciphrd committed:
✨ button to edit article on the article page

ciphrd committed:
♻️ improve UserGuard to only render if allowed

ciphrd committed:
✨ articles can be edited

ciphrd committed:
✨ add revisions to article informations

ciphrd committed:
✨ edition-in-progress indication banner

ciphrd committed:
✨ don't show minted editions on profile draft

ciphrd committed:
🐛 fix autosave not sending already-minted boolean

ciphrd committed:
✨ show edit cancellation button on article page

ciphrd committed:
Merge pull request #323 from fxhash/feature/articles-edit

Articles edition

maerzhase committed:
Merge branch 'feature/articles' into fix/#232-add-shortcut-for-creating-links

GemN committed:
fix: aria-label mismatch search

ciphrd committed:
♻️ include fonts & Noto -> Source Sans

ciphrd committed:
Merge branch 'feature/articles' of https://github.com/fxhash/website into feature/articles

GemN committed:
feat(article): add video media, refactor: handle of medias / folder architecture

GemN committed:
fix: clean console.log

GemN committed:
fix: figure name (semantics)

GemN committed:
fix: type updated correctly

GemN committed:
feat(article): add more highlight to code (plain, c, c++, julia, py, r, rust), refacto(article): file architecture of article code

GemN committed:
Merge branch 'feature/articles' of github.com:fxhash/fxhash-website into feature/articles-refacto

GemN committed:
refactor(article): simple folder structure, add guidelines to create plugins and new elements

GemN committed:
style(article): correct naming

GemN committed:
fix(article): project build correctly, fix reference error ssr prism

GemN committed:
refactor: optimize first load js for article

ciphrd committed:
Merge pull request #263 from fxhash/fix/#232-add-shortcut-for-creating-links

Fix/#232 add shortcut for creating links and allow configuration of inline menu via blockDefinitions

ciphrd committed:
Merge branch 'feature/articles' into feature/articles-table-normalize

ciphrd committed:
Merge pull request #287 from fxhash/feature/articles-table-normalize

feat(articles): normalisation of tables (handle bad formatted table and deletion of cells)

ciphrd committed:
Merge branch 'feature/articles' into feature/articles-video

ciphrd committed:
🐛 fix remaining merge line

ciphrd committed:
Merge pull request #334 from fxhash/feature/articles-video

feat(article): add video media, refactor: handle of medias / folder architecture

ciphrd committed:
Merge branch 'feature/articles' into feature/articles-code

ciphrd committed:
Merge pull request #335 from fxhash/feature/articles-code

feat(article): add more highlight to code (plain, c, c++, julia, py, r, rust), refacto(article): file architecture of article code

ciphrd committed:
Merge branch 'feature/articles' into feature/articles-refacto

ciphrd committed:
🚨 fix compiler warning

ciphrd committed:
Merge pull request #336 from fxhash/feature/articles-refacto

refactor(article): simple folder structure, add guidelines to create plugins and new elements

ciphrd committed:
♻️ implement changes made to tezos-storage-pointer spec

ciphrd committed:
Merge pull request #339 from fxhash/feature/articles-tezos-storage-pointers-v02

Implement changes made to tezos-storage-pointer spec

ciphrd committed:
✨ add mentions on the Generative Token page

ciphrd committed:
Merge pull request #341 from fxhash/feature/articles-token-mentions

Add mentions on the Generative Token page

ciphrd committed:
Merge pull request #343 from fxhash/feature/articles

Feature/articles -> dev

GemN committed:
fix(import): csv parser trim cols correctly

GemN committed:
fix(import): add scrollable import table

GemN committed:
fix(import): display number of address to imported, reduce size of import button to have a bigger list

GemN committed:
feat(article): add mentions to article

maerzhase committed:
add normalisation for emtpy links and without url

orphan links that have no text are removed entirely and links
that are created without url are being unwrapped.

- fix #326

ciphrd committed:
🐛 user guard shouldn't check if user was fetched

ciphrd committed:
📝 add articles contract address to the doc

GemN committed:
remove test files

ciphrd committed:
🚑 NFTArticleEditor dynamic import breaks reference on parent component

ciphrd committed:
Merge pull request #348 from fxhash/fix/remove-test-files

fix: remove test files

ciphrd committed:
✨ banner link to article text

ciphrd committed:
✨ added twitte rmeta tags

ciphrd committed:
Merge branch 'dev' into staging

ciphrd committed:
🐛 article URL id if slug not working

ciphrd committed:
Merge branch 'dev' into staging

ciphrd committed:
🔒️ sanitize href links

ciphrd committed:
✨ moderation UI to moderate an article

ciphrd committed:
✨ show a flag banner on the article page

ciphrd committed:
🚸 retain the route /explore/articles in the DOM

ciphrd committed:
✨ rules applied to articles based on moderation state

ciphrd committed:
✨ hidden articles are not shown on the user's profile anymore

ciphrd committed:
✨ update article page title

ciphrd committed:
🔧 env articles

ciphrd committed:
Merge pull request #351 from fxhash/feature/articles-moderation

Article moderation tools

ciphrd committed:
💄 fix compiler warnings

GemN committed:
fix(user): articles load correctly without infinite loop

GemN committed:
fix(article): remove entire nodes when deleting foward/backward for medias (figure) and table

ciphrd committed:
Merge pull request #358 from fxhash/fix/articles-user-loading

fix(user): articles load correctly without infinite loop

ciphrd committed:
Merge pull request #359 from fxhash/fix/articles-image-crash

fix(article): remove entire nodes when deleting foward/backward for medias (figure) and table

ciphrd committed:
Merge branch 'dev' into fix/normalisation-for-links

ciphrd committed:
Merge pull request #346 from fxhash/fix/normalisation-for-links

Adds normalisation for links

ciphrd committed:
🐛 add missing thumbnail caption

GemN committed:
fix(article): uncomment autosave

GemN committed:
fix(article): blockquote style work in preview

GemN committed:
fix(article): add break in a list only convert to paragraph at the end of the list

ciphrd committed:
💄 styling for user mentions more readable

GemN committed:
fix(article): invalid slug with just a dot redirect to id

GemN committed:
fix(article): change mention style / remove rehypeFormat that was badly formatting custom components

GemN committed:
fix(article): change font weight of mentions

ciphrd committed:
🔒️ add headers to protect from embedding fxhash into an iframe / popup

ciphrd committed:
🔒️ stricter CSP to prevent anything not fxhash

ciphrd committed:
🔒️ improve security headers, support for articles too

ciphrd committed:
✨ update the mint reserve input (pack with method id)

maerzhase committed:
fix date floating for mobile article cards

ciphrd committed:
✨ add basic reserve mint pass support

ciphrd committed:
✨ add mint pass unpack reserve

GemN committed:
Merge pull request #360 from fxhash/fix/autosave

fix(article): uncomment autosave

ciphrd committed:
Merge branch 'dev' into feature/article-mentions

ciphrd committed:
Merge pull request #345 from fxhash/feature/article-mentions

feat(article): add mentions to article

ciphrd committed:
💄 articles > explore remove top level padding + responsive margin between articles

ciphrd committed:
Merge pull request #367 from fxhash/fix/#353-article-card-responsive

Fix/article: card responsive

ciphrd committed:
Merge branch 'dev' into fix/get-article-url

ciphrd committed:
Merge pull request #365 from fxhash/fix/get-article-url

fix(article): invalid slug with just a dot redirect to id

ciphrd committed:
Merge pull request #363 from fxhash/fix/article-list-break

fix(article): add break in a list only convert to paragraph at the end of the list

ciphrd committed:
💄 remove top and bottom margin on first/last elements inside blockquotes

ciphrd committed:
Merge pull request #362 from fxhash/fix/article-blockquote-style

fix(article): blockquote style work in preview

ciphrd committed:
Merge pull request #344 from fxhash/fix/import-csv

hotfix(import): csv parser trim space in cols / handle long size

ciphrd committed:
✨ added article actions to list of actions

GemN committed:
fix(article): update media size (video, image)

ciphrd committed:
🔧 10000 -> 50000 max editions

maerzhase committed:
add article infos to user actions

maerzhase committed:
add components for article related actions

GemN committed:
feat(live-minting): add minimalist layout, page generative, page reveal, connect by wallet for live-minting

GemN committed:
feat(live-minting): add live event skeleton

ciphrd committed:
♻️ refacto of the actions

ciphrd committed:
Merge pull request #373 from fxhash/fix/#369-add-article-related-actions

Fix/#369 add article related actions

ciphrd committed:
Merge branch 'staging' into dev

ciphrd committed:
🚨 fix wront imports

ciphrd committed:
🚨 HARDFIX compiler error - bad, bad :(

ciphrd committed:
🔧 added rehype-format back for processing documentation md

GemN committed:
fix(article): autosave warn user if using incognito mode, check that an article saved correctly, check localStorage is available

GemN committed:
fix(article): autosave check size used

maerzhase committed:
add new tabstyle subtabs-vertical

- related to #376

maerzhase committed:
add iconSide null for centered icon button

maerzhase committed:
add query for user articles owned

maerzhase committed:
add interface NFTArticlesOwned

maerzhase committed:
add prop for hiding the spacing above the section content

maerzhase committed:
add articles to user collection

adds the articles owned to the user collection view. articles
can be accessed via subtab-vertical tabs. objkts collection is
found on the collection/index route while articles are being served
under collection/articles. articles are rendered in plain list
without any filtering and sorting options. the enjoy button was
moved into the toolbar for searching the objekts collection.

- fix #376

GemN committed:
fix(article): embed youtube need allow-presentation to work properlly

ciphrd committed:
🔒️ limit user search input to users with safe flag

ciphrd committed:
💄 small adjustements to vertical subtabs style

ciphrd committed:
💄 articles are centered like for the rest of the layout

ciphrd committed:
📱 select font size smaller on mobile

ciphrd committed:
💄 updated enjoy collection btn icon & less padding on single icon btn

ciphrd committed:
♻️ user collec > refacto split sub comps

ciphrd committed:
✨ show number of editions owned on user collec articles

ciphrd committed:
Merge pull request #380 from fxhash/feature/#376-add-articles-to-user-collection

Feature/marketplace-v3:  Add articles to user collection

ciphrd committed:
💄 live minting UI adjustement

ciphrd committed:
✨ live minting > show wallet balance at the top

ciphrd committed:
💄 adjust live minting token card design

GemN committed:
feat(article): add marketplace listings to article

GemN committed:
fix(article): change article li style

ciphrd committed:
🔒️ auth flow for the event (check id & token validity)

ciphrd committed:
✨ live minting show event projects

ciphrd committed:
💄 wallet balance increase tezos size

ciphrd committed:
Merge branch 'feature/reserve-mint-pass' into feature/live-minting

ciphrd committed:
🔒️ show mint button if reserve mint pass is met

ciphrd committed:
✨ full support for minting with reserve mint pass

ciphrd committed:
💄 minor UI changes to the article bottom infos

ciphrd committed:
✨ removed linking of the tabs (created a weird UX experience)

ciphrd committed:
Merge pull request #382 from fxhash/feature/marketplace-listings

feat(article): add marketplace listings to article

ciphrd committed:
✨ added actions for marketplace v3 (only articles supported)

ciphrd committed:
✨ added contract operations for marketplace v3 + btns for articles

ciphrd committed:
✨ add quick collect button on article

ciphrd committed:
🔇 clear console logs

ciphrd committed:
Merge pull request #384 from fxhash/feature/marketplace-v3-contracts

Article contract interactions + Actions for marketplace V3

ciphrd committed:
🔧 set marketplace v3 address in env

ciphrd committed:
Merge branch 'feature/marketplace-v3' into dev

ciphrd committed:
📝 added address contract doc

ciphrd committed:
Merge branch 'dev' into feature/live-minting

ciphrd committed:
✨ naan url & log error

ciphrd committed:
🐛 fix user by pkh > collection > articles 404

GemN committed:
fix(article): media video dont crash page when url is wrong

maerzhase committed:
Merge branch 'dev' into devx/#97-add-eslint-prettier

maerzhase committed:
add prettier configuration

maerzhase committed:
add lint.fix command

ciphrd committed:
✨ live event, improved onboarding flow

ciphrd committed:
🚸 improve user experience for live minting flow

ciphrd committed:
✨ handle minting pass expired error

ciphrd committed:
✨ hide warning on live minting context

ciphrd committed:
Merge branch 'feature/live-minting' into dev

ciphrd committed:
✨ display mint pass group details in the reserve

ciphrd committed:
🔧 added events url to the main .env

ciphrd committed:
✨ added fetch event when input reserve mint pass

ciphrd committed:
🐛 add support for reserve mint pass at preview

gregsadetsky committed:
fix: small typos in docs

GemN committed:
fix(article): medias, thumbnail and tezos storage take viewport as maximum height

GemN committed:
fix(article): thumbnail update correctly, don't throw error on edition

GemN committed:
fix(article): card nft article have buttons working as intended -> edit goes to article edit page, bin delete the local changes

ciphrd committed:
Merge branch 'dev' into devx/#97-add-eslint-prettier

ciphrd committed:
✨ format code; run `run lint:fix`

ciphrd committed:
🚨 fix compiler warnings (@ts-ignore were move by auto format)

ciphrd committed:
Merge pull request #121 from fxhash/devx/#97-add-eslint-prettier

Add eslint plugin prettier

ciphrd committed:
Revert "Add eslint plugin prettier"

ciphrd committed:
Merge pull request #397 from fxhash/revert-121-devx/#97-add-eslint-prettier

Revert "Add eslint plugin prettier"

GemN committed:
fix: optimize infinite scroll - doesn't trigger empty queries when under 20 results

ciphrd committed:
Merge pull request #372 from fxhash/fix/article-media-size

fix(article): update media size (video, image)

ciphrd committed:
Merge pull request #379 from fxhash/fix/articles-autosave

fix(article): autosave warn user if using incognito mode, check that an article saved correctly, check localStorage is available

ciphrd committed:
Merge pull request #381 from fxhash/fix/article-embed-sandbox

fix(article): embed youtube need allow-presentation to work properlly

ciphrd committed:
Merge pull request #383 from fxhash/fix/article-li-style

fix(article): change article li style

ciphrd committed:
Merge pull request #388 from fxhash/fix/article-medias-url

fix(article): media video dont crash page when url is wrong

ciphrd committed:
Merge pull request #393 from fxhash/fix/article-edit-thumbnail-error

fix(article): thumbnail update correctly, don't throw error on edition

ciphrd committed:
Merge pull request #391 from gregsadetsky/bug/small-typos

fix(article): fix small typos

ciphrd committed:
🐛 remove detect incognito

ciphrd committed:
Merge branch 'dev' of https://github.com/fxhash/website into dev

ciphrd committed:
Merge branch 'dev' into fix/article-card-edit

ciphrd committed:
Merge pull request #394 from fxhash/fix/article-card-edit

fix(article): card nft article have buttons working as intended -> edit goes to article edit page, bin delete the local changes

ciphrd committed:
Merge pull request #398 from fxhash/fix/optimize-infinite-scroll

fix: optimize infinite scroll - doesn't trigger empty queries when under 20 results

ciphrd committed:
🚨 fix compiler warnings

ciphrd committed:
🔧 added eslint config back

ciphrd committed:
🎨 run lint:fix

ciphrd committed:
🚨 fix post lint ts-ignore merge magic

ciphrd committed:
Merge pull request #399 from fxhash/fix/prettier-back

Bring prettier back due to improper merge

maerzhase committed:
Merge branch 'dev' into fix/#304-softbreak-plugin
